### PR TITLE
feat(1785): Add groupEventId to event create [3]

### PR DESCRIPTION
**Pull request by tkyi**
## Context

When users restart builds in events, it's not always clear which event to restart failed builds from. Due to this confusion, the join build doesn't always start as expected.
This logic can be made simpler by grouping all events with the same parent.

## Objective

This PR adds `groupEventId` to events during creation to allow for easier grouping. When an event is not restarted (ie is original), it will not have a `groupEventId`.

## References

Related to #1785 
Blocked by https://github.com/screwdriver-cd/models/pull/440/files

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/events/create.js [hunk 1]
```diff
@@ -39,11 +39,23 @@ module.exports = () => ({
...
-                                    parentEventId: b.eventId
+                                    parentEventId: b.eventId,
+                                    groupEventId: request.payload.groupEventId,
+                                    parentBuilds: b.parentBuilds
                                 };
 
-                                if (b.parentBuilds) {
-                                    restartConfig.parentBuildsInfo = b.parentBuilds;
+                                if (!restartConfig.parentBuilds) {
+                                    restartConfig.parentBuilds = request.payload.parentBuilds;
```
Comment (klu909) [inline plugins/events/create.js]: does this require ui changes?

Comment (tkyi) [reply plugins/events/create.js]: should be the same logic as before, just refactored

File: plugins/builds/index.js [hunk 2]
```diff
@@ -267,17 +274,18 @@ async function createEvent(config) {
...
- * @param  {Number}   config.parentBuildId      Parent Build Id
+ * @param  {Number}   config.parentBuildId      Parent Build ID
  * @param  {Object}   config.parentBuilds       Builds that triggered this build
  * @param  {String}   config.causeMessage       Cause message of this event
- * @param  {Boolean}  [config.start]            Whether to start the build after creating
+ * @param  {Number}   [config.parentEventId]    Parent event ID
+ * @param  {Number}   [config.groupEventId]     Group parent event ID
  * @return {Promise}
  */
 async function createExternalBuild(config) {
```
Comment (jithine) [inline plugins/builds/index.js]: how about `createInternalBuild` function ? how is `groupEventId` handled there  ?

Comment (tkyi) [reply plugins/builds/index.js]: `groupEventId` is a field for an `event`. `createExternalBuild` creates a new `event` (with subsequent builds), but `createInternalBuild` creates a single `build`, so there's no where to set the `groupEventId`.

**Pull request merged**